src/api/middleware/company.js

```diff
@@ -85,6 +85,45 @@ export const profileComplete = async (req, res, next) => {
     return next();
 };
 
+export const canAccessProfile = (companyId) => async (req, res, next) => {
+    const company = await new CompanyService().findById(companyId, true);
+
+    const notFound = () =>
+        new APIError(
+            HTTPStatus.UNPROCESSABLE_ENTITY,
```

Comment:
Shouldn't this be "NOT_FOUND" ?

Reply:
Same reason as above, after #282 it would be NOT_FOUND.